src/main/java/com/ibm/crypto/plus/provider/AESKey.java

```diff
@@ -35,7 +34,7 @@ final class AESKey implements SecretKey, Destroyable {
      * @exception InvalidKeyException
      *                if the given key has wrong size
      */
-    public AESKey(byte[] key) throws InvalidKeyException {
+    AESKey(byte[] key) throws InvalidKeyException {
```

Comment:
We removed it, because the constructor shouldn't be called by external classes. I updated those visibilities based on the warnings I was getting, so there might be other visibilities that need to change. I'm not sure whether we should update them all as part of this PR, but let me know which ones you've seen @taoliult. What do you think @jasonkatonica?

Reply:
You mean, all the classes constructors in the `com/ibm/crypto/plus/provider/ `package (Except the OpenJCEPlus/OpenJCEPlusFIPS provider class) shouldn't be called by external classes?

For example, `AESCipher` class, its constructor is still public. And the `EdDSAKeyFactory` class, its constructor changed from `protected` to `public`.

I am totally ok with not update them all in this PR, and if all the classes constructors in the `com/ibm/crypto/plus/provider/ `package shouldn't be called by external classes, then maybe we can have another PR to update them for the consistent?

Reply:
Hmm, I don't remember why I changed the constructor for `EdDSAKeyFactory`. Maybe they should be `protected`. Let me check it out.  

As far as the `AESCipher`, it didn't cause any warnings, so I didn't touch, and there might be other similar issues in the code actually.

Reply:
I updated the visibility of the constructors of `EdDSAKeyFactory`. Should be fine now.